specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+                    handleRemove === undefined
+                      ? undefined
+                      : (): void => handleRemove(categoryIndex, itemIndex)
+                  }
+                />
+              );
+            })}
+          </div>
+          {isEditing && (
+            <Button.LikeLink
+              onClick={(): void => goToEditingState(categoryIndex)}
+            >
+              <span className={className.dataEntryAdd}>{icons.plus}</span>
+              {commonText('add')}
+            </Button.LikeLink>
+          )}
+        </div>
+      ))}
+    </>
+  );
+}
+
+function useDefaultLayout(
+  statsSpec: IR<
+    IR<{
+      readonly label: string;
+      readonly items: StatCategoryReturn | undefined;
+    }>
+  >
+): StatLayout {
+  return React.useMemo(
+    () =>
+      Object.entries(statsSpec).map(([pageName, pageStatsSpec]) => ({
+        label: pageName,
+        categories: Object.entries(pageStatsSpec).map(
+          ([categoryName, { label, items }]) => ({
+            label,
+            items: Object.entries(items ?? {}).map(([itemName]) => ({
+              type: 'DefaultStat',
+              pageName: pageName,
+              categoryName,
+              itemName,
+            })),
+          })
+        ),
+      })),
+    [statsSpec]
+  );
+}
+
+function useCustomStatQueryBuilderSpec(queryId: number):
+  | {
+      readonly tableName: keyof Tables;
+      readonly fields: RA<
+        Partial<SerializedResource<SpQueryField>> & { readonly path: string }
+      >;
+      readonly label: string;
+    }
+  | undefined {
+  const [data] = useAsyncState(
+    React.useCallback(
+      async () =>
+        fetchResource('SpQuery', queryId).then((queryData) => ({
+          tableName: queryData.contextName as keyof Tables,
+          fields: queryData.fields.map((field) => ({
+            ...field,
+            path: QueryFieldSpec.fromStringId(
+              field.stringId,
+              field.isRelFld ?? false
+            )
+              .toMappingPath()
+              .join('.'),
+          })),
+          label: queryData.name,
+        })),
+      [queryId]
+    ),
+    false
+  );
+  return data;
+}
+
+function DefaultStatItem({
+  statsSpec,
+  pageName,
+  categoryName,
+  itemName,
+  onRemove: handleRemove,
+  onClick: handleClick,
+}: {
+  readonly statsSpec: IR<
+    IR<{
+      readonly label: string;
+      readonly items: StatCategoryReturn;
+    }>
+  >;
+  readonly pageName: string;
+  readonly categoryName: keyof typeof statsSpec;
+  readonly itemName: string;
+  readonly onRemove: (() => void) | undefined;
+  readonly onClick: (() => void) | undefined;
+}): JSX.Element {
+  const statSpecItemPage = statsSpec[pageName];
+  const statSpecItemObject = statSpecItemPage[categoryName]?.items;
+  const statSpecItem =
+    statSpecItemObject === undefined ? undefined : statSpecItemObject[itemName];
+  const statValue =
+    statSpecItem === undefined ? undefined : statSpecItem.spec.type ===
```

Comment:
Could simplify this by replacing `undefined` here with `<p>{commonText('loading')}</p>` and then replace `const statValue =` with `return`